installing_on_kubernetes/_topics/migrate_from_appliance.md

```diff
@@ -0,0 +1,131 @@
+## Preparing the Kubernetes namespace
+The deploy directory referenced below can be found in [the manageiq-pods repo](https://github.com/ManageIQ/manageiq-pods/tree/master/manageiq-operator/deploy)
+
+1. Search for the Custom Resource Definition (CRD) and create it if it doesn't already exist.
+```bash
+oc get crds | grep manageiqs.manageiq.org
+```
+```bash
+oc create -f deploy/crds/manageiq.org_manageiqs_crd.yaml
+```
+
+2. Set up RBAC.
+```bash
+oc create -f deploy/role.yaml
+oc create -f deploy/role_binding.yaml
+oc create -f deploy/service_account.yaml
+```
+
+3. Deploy the operator in your namespace.
+```bash
+$ oc create -f deploy/operator.yaml
+```
+
+## Migrating from Appliances
+
+### Collect data from the appliance
+1. Take a backup of the database
+```bash
+pg_dump -Fc -d vmdb_production > /root/pg_dump
+```
+
+2. Export the encryption key and Base64 encode it for Openshift
+```bash
+vmdb && rails r "puts Base64.encode64(ManageIQ::Password.v2_key.to_s)"
+```
+
+3. Get the region number
+```bash
+vmdb && rails r "puts MiqRegion.my_region.region"
+```
+
+4. Get the GUID of the server that you want to run as.
+```bash
+vmdb && cat GUID
+```
+
+### Restore the backup into the kubernetes environment
+1. Create a YAML file defining the Custom Resource (CR). Minimally you'll need the following:
+```yaml
+apiVersion: manageiq.org/v1alpha1
+kind: ManageIQ
+metadata:
+  name: miq
+spec:
+  applicationDomain: <your application domain name>
+  databaseRegion: <region number from the appliance above>
+  serverGuid: <GUID value from appliance above>
+```
+
+2. Create the CR in your namespace. Once created, the operator will create several additional resource and start deploying the app.
+```bash
+oc create -f <file name from above>
+```
+
+3. Edit the app secret inserting the encryption key from the appliance. Replace the "encryption-key" value with the value we exported from the appliance above.
```

Comment:
Can this be done as step one?  That was the app-secrets already exists?

Reply:
The operator creates the secret after Step 2.  Alternatively we could manually define the secret, but I think this might be easier until this is fully automated.

Reply:
yeah i was thinking manually define the secret (or offer a yaml that people could edit, then apply)